ReactWindows/ReactNative.Shared/Bridge/ReflectionReactDelegateFactory.cs

```diff
@@ -163,20 +170,34 @@ private Extractor CreateExtractor(Type type, string moduleName, string methodNam
             var args = new object[extractors.Count];
             for (var j = 0; j < c; ++j)
             {
-                var result = extractors[j](reactInstance, jsArguments, idx);
-                args[j] = result.Value;
-                idx = result.NextIndex;
+                var extractorResult = extractors[j](reactInstance, jsArguments, idx);
+                args[j] = extractorResult.Value;
+                idx = extractorResult.NextIndex;
             }
 
-            if (genericDelegate != null)
+            var hasDelegate = (genericDelegate != null) && ReferenceEquals(moduleInstance, genericDelegate.DelegateTarget);
+            if (method.ReturnType == typeof(void))
             {
-                genericDelegate.Invoke(args);
+                if (hasDelegate)
+                {
+                    genericDelegate.Invoke(args);
+                }
+                else
+                {
+                    // This should only happen for React methods with greater than 16 arguments.
+                    method.Invoke(moduleInstance, args);
+                }
+
+                return null;
             }
-            else
+
+            var result = hasDelegate ? genericDelegate.Invoke(args) : method.Invoke(moduleInstance, args);
+            if (result == null)
             {
-                // This should only happen for React methods with greater than 16 arguments.
-                method.Invoke(moduleInstance, args);
+                return new JValue((object)null);
             }
+
+            return JToken.FromObject(result);
```

Comment:
>JToken.FromObject(result); [](start = 19, length = 26)

To minimize impact on the implementation, let's just return a static field storing `JValue.CreateUndefined()` in case of void returning methods.